tools/submit_awcy.py

```diff
@@ -78,12 +79,20 @@ def f(*popenargs, **kwargs):
 if args.prefix is None:
     args.prefix = args.branch
 
-commit = subprocess.check_output("git rev-parse HEAD", shell=True).strip()
-short = (
-    subprocess.check_output("git rev-parse --short HEAD", shell=True)
-    .strip()
-    .decode("utf-8")
-)
+if args.commit is None:
+    commit = subprocess.check_output("git rev-parse HEAD", shell=True).strip()
+    short = (
+        subprocess.check_output("git rev-parse --short HEAD", shell=True)
+        .strip()
+        .decode("utf-8")
+    )
+else:
+    commit = args.commit
+    short_arg = "git rev-parse --short " + args.commit
+    short = (
+        subprocess.check_output(short_arg, shell=True).strip().decode("utf-8")
+    )
+
```

Comment:
This could be extracted down from both branches

Reply:
I tried to address this, can you please check agian